### PR TITLE
Match Briefcase export behavior when repeats are submitted to Google Sheets

### LINKED ISSUE #2063: Match Briefcase export behavior when repeats are submitted to Google Sheets
#### Software and hardware versions 
Collect v1.14.0 beta

#### Problem description
Currently repeats are tied to their parent record but not their parent repeat in the case of nested repeats.

#### Steps to reproduce the problem
Use a form with nested repeats with Google Sheets as the server - [nested-indexed-repeat.xml.txt](https://github.com/opendatakit/collect/files/1849095/nested-indexed-repeat.xml.txt) for example.

#### Expected behavior
Match the Briefcase behavior by specifying `KEY` and `PARENT_KEY` columns instead of the current `instanceID` column that all repeats get. The only difference is that `KEY` includes the XPath path of the current node and `KEY_PARENT` includes the path of the parent node.

| friend_index | pet_index | friend | pet | note |              meta-instanceID              |                    KEY                    |
|--------------|-----------|--------|-----|------|-------------------------------------------|-------------------------------------------|
|            1 |         2 | Bob    | Baz |      | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6 |

| friend_count | friend_name | pet_total | pets_count |                PARENT_KEY                 |                         KEY                          |
|--------------|-------------|-----------|------------|-------------------------------------------|------------------------------------------------------|
|            1 | Bob         |         2 |          2 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[1] |
|            2 | Bill     |         1 |          1 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[2] |
|            3 | Farouk     |         2 |          2 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6 | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[3] |

| pet_count | pet_name |                      PARENT_KEY                      |                             KEY                              |
|-----------|----------|------------------------------------------------------|--------------------------------------------------------------|
|         1 | Fido      | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[1] | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[1]/pets[1] |
|         2 | Fanny      | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[1] | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[1]/pets[2] |
|         1 | Prancer     | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[2] | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[2]/pets[1] |
|         1 | Dancer       | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[3] | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[3]/pets[1] |
|         2 | Bob       | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[3] | uuid:3cc27b08-7ab4-42a6-9261-48eb2f9827d6/friends[3]/pets[2] |

### PR DESCRIPTION
**Pull request by grzesiek2010**
Closes #2063 

#### What has been done to verify that this works as intended?
I've tested a form with `instanceID` and without (both with nested repeatable groups).

#### Why is this the best possible solution? Were any other approaches considered?
I added two columns to identify repeatable groups (`PARENT_KEY` and `KEY`) like in Briefcase.

#### Are there any risks to merging this code? If so, what are they?
I think it's not risky but a few different forms should be tested to be sure everything is ok.

#### Do we need any specific form for testing your changes? If so, please attach one.
I used:
[nRepeatsWithoutInstanceID.xml.txt](https://github.com/opendatakit/collect/files/1850844/nRepeatsWithoutInstanceID.xml.txt)
[nRepeats.xml.txt](https://github.com/opendatakit/collect/files/1850845/nRepeats.xml.txt)

### CODE REVIEW

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java [hunk 1]
```diff
@@ -371,24 +379,33 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
                 }
             }
         }
+        if (element.isRepeatable()) {
+            answers.put(PARENT_KEY, parentKey);
+        }
+        if (hasRepeatableGroups(element)) {
+            answers.put(KEY, key);
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java]: The `KEY` is useful for reconstructing order even when there aren't further nested groups. Also, including `KEY` for all repeats matches Briefcase. So this should go in the prior `if`.

File: collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java [hunk 2]
```diff
@@ -371,24 +379,33 @@ private void createSheetsIfNeeded(TreeElement element) throws UploadException {
...
-            columnTitles.add(0, INSTANCE_ID);
-        }
         for (TreeElement child : getChildElements(element)) {
             columnTitles.add(getElementTitle(child));
         }
+        if (element.isRepeatable()) {
+            columnTitles.add(PARENT_KEY);
+        }
+        if (hasRepeatableGroups(element)) {
+            columnTitles.add(KEY);
```
Comment (lognaturel) [inline collect_app/src/main/java/org/odk/collect/android/tasks/InstanceGoogleSheetsUploader.java]: Same as above. Let's include the `KEY` for all repeats.

**Comment by lognaturel** [comment]
I know @grzesiek2010 is asleep so I made the quick change and this is ready for a final review and hopefully merge.

**Review by lognaturel — APPROVED** [review]

**Comment by mmarciniak90** [comment]
Tested with success as a part of https://github.com/opendatakit/collect/pull/2066

Verified on Android: 4.1, 4.4, 6.0, 7.0, 8.0

Verified cases: 
* nested-indexed-repeat.xml
* nRepeatsithoutInstanceID.xml
* nRepeats.xml
* fewRepeatGroupsGoogleSheets.xml

@opendatakit-bot label "behavior verified"

**Pull request merged**